### PR TITLE
Add beolink_join action source_id parameter to Bang & Olufsen 

### PR DESCRIPTION
**Pull request by mj23000**
## Proposed change

Add new "source_id" parameter to the existing `beolink_join` custom action.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/132377
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Updated the Bang & Olufsen integration documentation to include a new optional attribute `source_id` for the `bang_olufsen.beolink_join` action.
	- Clarified compatibility of `source_id` values with specific hardware platforms.
	- Enhanced clarity of action data attributes for the integration.
	- Improved overall structure and formatting for better readability.

### CODE REVIEW

File: source/_integrations/bang_olufsen.markdown [hunk 1]
```diff
@@ -259,9 +259,10 @@ The Bang & Olufsen integration additionally supports different custom actions
 
 Join a Beolink experience.
 
-| Action data attribute | Optional | Description                           |
-| --------------------- | -------- | ------------------------------------- |
-| `beolink_jid`         | yes      | Manually specify Beolink JID to join. |
+| Action data attribute | Optional | Description                                                                                                                                                                                    |
+| --------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `beolink_jid`         | yes      | Manually specify Beolink JID to join.                                                                                                                                                          |
+| `source_id`           | yes      | Specify which source to join, behavior varies between platforms. Source names prefaced by a platform name can only be used when connecting to that platform. A defined Beolink JID is required |
```
Comment (andrewsayre) [inline source/_integrations/bang_olufsen.markdown]: Once this [comment is resolved](https://github.com/home-assistant/core/pull/132377#discussion_r1871808853), I'd like to see the possible values that can be supplied enumerated in the documentation.

This is confusing to me:
> Source names prefaced by a platform name can only be used when connecting to that platform.

Is that talking about a HA platform (i.e. `media_player`) or something else? Let's clarify what this is talking about and provide an example.

**Review by andrewsayre — CHANGES_REQUESTED** [review]

File: source/_integrations/bang_olufsen.markdown (hunk 1 above)
Comment (mj23000) [reply source/_integrations/bang_olufsen.markdown:265]: Platform refers to the Bang & Olufsen hardware platform, ASE, Mozart etc.

Comment (mj23000) [reply source/_integrations/bang_olufsen.markdown]: This is about hardware platforms used by Bang & Olufsen. ASE, Mozart and the Beolink converter. This should be clarified.

File: source/_integrations/bang_olufsen.markdown [hunk 2]
```diff
@@ -259,9 +259,10 @@ The Bang & Olufsen integration additionally supports different custom actions
...
+| --------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `beolink_jid`         | yes      | Manually specify Beolink JID to join.                                                                                                                                                                                                                                                                                                                |
+| `source_id`           | yes      | Specify which source to join, behavior varies between hardware platforms. Source names prefaced by a platform name can only be used when connecting to that platform. For example "ASE Beoradio" can only be used when joining an ASE device, while ”ASE / Mozart Deezer” can be used with ASE or Mozart devices. A defined Beolink JID is required. |
```
[the remark this answers is not in the document]
Comment (andrewsayre) [reply source/_integrations/bang_olufsen.markdown:265]: I agree with with the bot. 😄 Please add the following:
1. A table of allowed values that describes the hardware platforms that support each option
2. A service usage example

Thank you!

**Review by andrewsayre — CHANGES_REQUESTED** [review]
Just two small things!

**Review by andrewsayre — APPROVED** [review]

**Pull request merged**